### PR TITLE
Fix server adapters

### PR DESCRIPTION
**Pull request by connorcarpenter**

### CODE REVIEW

File: adapters/bevy/shared/src/flag.rs [hunk 1]
```diff
@@ -3,10 +3,9 @@ pub struct Flag {
 }
 
 impl Flag {
+    #![allow(clippy::new_without_default)]
```
Comment (Shatur) [inline adapters/bevy/shared/src/flag.rs:6]: But why just not use default trait?

Comment (Shatur) [reply adapters/bevy/shared/src/flag.rs:6]: The warning won't be triggered in case of only `default`.

Comment (connorcarpenter) [reply adapters/bevy/shared/src/flag.rs:6]: Well for some reason even when I was using the default trait, it built on my machine but then wouldn't build in the CI ! The compiler said "you need to implement Default" when it clearly WAS implemented. So moving to a plain new() method got it to compile, but then Clippy complains that it should be default :man_facepalming: ...

Comment (Shatur) [reply adapters/bevy/shared/src/flag.rs:6]: It worked before, I already fixed this warning by replacing `new` with `Default` macro.

Comment (Shatur) [reply adapters/bevy/shared/src/flag.rs:6]: You can see in the PR diff that you removed `Default` macro.

Comment (connorcarpenter) [reply adapters/bevy/shared/src/flag.rs:6]: Sorry sorry, it wasn't the CI that wouldn't compile, it was `cargo publish` that for some reason complained about Default not being implemented.

You know what.. I'm going to revert this back and leave fixing this properly for later :P

Comment (Shatur) [reply adapters/bevy/shared/src/flag.rs:6]: Oh, that strange :)

File: adapters/bevy/server/src/resource.rs [hunk 2]
```diff
@@ -1,6 +1,15 @@
+use std::default::Default;
+
 use naia_bevy_shared::Flag;
 
-#[derive(Default)]
 pub struct ServerResource {
     pub ticker: Flag,
 }
+
+impl Default for ServerResource {
```
Comment (Shatur) [inline adapters/bevy/server/src/resource.rs:9]: You probably can use macro for this one too.

File: adapters/bevy/client/src/resource.rs [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
 use naia_bevy_shared::Flag;
 
-#[derive(Default)]
 pub struct ClientResource {
     pub ticker: Flag,
     pub connector: Flag,
     pub disconnector: Flag,
 }
+
+impl Default for ClientResource {
```
Comment (Shatur) [inline adapters/bevy/client/src/resource.rs:11]: Sorry, you probably forgot to update this one :D

File: adapters/bevy/server/src/resource.rs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+use std::default::Default;
+
```
Comment (Shatur) [inline adapters/bevy/server/src/resource.rs:2]: Probably an extra `use`?

File: demos/bevy/client/src/resources.rs [hunk 5]
```diff
@@ -1,3 +1,5 @@
+use std::default::Default;
```
Comment (Shatur) [inline demos/bevy/client/src/resources.rs:1]: And here.

**Pull request merged**